lib/cloudant.js

```diff
@@ -879,41 +879,54 @@ Cloudant.prototype.selectModel = function(model, migrate) {
 
 /**
  * Perform autoupdate for the given models. It basically calls db.index()
- *
+ * It does NOT destroy previous model instances if model exists, only
+ * `automigrate` does that.
  * @param {String[]} [models] A model name or an array of model names. If not
  * present, apply to all models
  * @param {Function} [cb] The callback function
  */
 Cloudant.prototype.autoupdate = function(models, cb) {
   debug('Cloudant.prototype.autoupdate %j', models);
-  this.automigrate(models, cb);
+  var self = this;
+  async.eachSeries(models, function(model, cb2) {
+    debug('Cloudant.prototype.autoupdate model %j', model);
+    var mo = self.selectModel(model, true);
+    self.updateIndex(mo, model, cb2);
```

Comment:
Please don't do this, as it's not necessary. The innermost definition of cb will be the one that gets used, so there's no need to differentiate between them unless you have special logic where you might call one of the callbacks, but not the other.

If that was the case, then I would definitely want a better naming convention than `cb2`. :/